### **_Importar librerias_**

In [2]:
import pandas as pd
import numpy as np

### **_Letura de datos_**

In [3]:
# Aqui leo el fichero de los pisos
df_131= pd.read_csv("dfpisosCompleto12012024.csv")

C:\Users\Lenovo\AppData\Local\Temp\ipykernel_18416\3328357953.py:2: DtypeWarning: Columns (4,14,15,16,20,21,22,24,30,37,40,43,74,78,81,82,83,85,86,87,88,89,90,91,93,94,96,100,101,103,106,107,110) have mixed types. Specify dtype option on import or set low_memory=False.
  df_131= pd.read_csv("dfpisosCompleto12012024.csv")


#### **_Configuracion extra_**

In [4]:
# Aqui ajusto el limite de despliegue de columnas para que se observen todas sin truncarse.
pd.set_option("display.max_columns",None)
# Aqui ajusto el limite de la informacion que se muestra en las casillas de los datos para que se muestre todo sin truncarse.
pd.set_option("display.max_colwidth", None)

### **_EDA/ Exploratory Data Analisys_**

In [5]:
# Selecciono las primeras 31 columnas. De la 0 a la 30.
df_131=df_131.iloc[:, :31]

In [6]:
# Checo la forma de mi dataframe y que realmente tome 31 columnas.
df_131.shape

(10241, 31)

In [7]:
# Muestreo de las primeras 5 filas 
df_131.head()

title  \
0                          Piso en venta en Madrid   
1  Piso en venta en Montecarmelo  El Goloso Madrid   
2                 Piso en venta en Malasana Madrid   
3             Piso en venta en Pueblo Nuevo Madrid   
4               Piso en venta en Villaverde Madrid   

                                                                                                 url  \
0                           https://www.pisosmadrid.com.es/propiedad/3863841-piso-en-venta-en-madrid   
1  https://www.pisosmadrid.com.es/propiedad/3865872-piso-en-venta-en-montecarmelo---el-goloso-madrid   
2                  https://www.pisosmadrid.com.es/propiedad/3865902-piso-en-venta-en-malasana-madrid   
3              https://www.pisosmadrid.com.es/propiedad/3865903-piso-en-venta-en-pueblo-nuevo-madrid   
4                https://www.pisosmadrid.com.es/propiedad/3866311-piso-en-venta-en-villaverde-madrid   

      precio  precio_anterior descuento  EUR/m2  m2_constr  \
0  2000000.0              NaN       NaN     NaN       54.0   
1   287000.0              NaN       NaN     NaN      142.0   
2   200000.0              NaN       NaN     NaN      170.0   
3   650000.0              NaN       NaN     NaN       75.0   
4   650000.0              NaN       NaN     NaN       60.0   

              distrito  cod_distrito        barrio  cod_barrio  \
0                  NaN           4.0           NaN        41.0   
1  Fuencarral-El Pardo          21.0     El Goloso       213.0   
2               centro           2.0           NaN        24.0   
3        Ciudad Lineal           5.0  Pueblo Nuevo        54.0   
4           Villaverde           5.0           NaN        54.0   

                      zona calle detalle consumoce_ano letrace emisiones_co2  \
0                      NaN   NaN     NaN           NaN     NaN           NaN   
1  Montecarmelo  El Goloso   NaN     NaN           NaN     NaN           NaN   
2                 Malasana   NaN     NaN           NaN     NaN           NaN   
3             Pueblo Nuevo   NaN     NaN           NaN     NaN           NaN   
4               Villaverde   NaN     NaN           NaN     NaN           NaN   

  tipologia  propiedad_completa estado amueblado planta habitaciones  \
0      Piso                 NaN    NaN       0.0    NaN          NaN   
1      Piso                 NaN    NaN       0.0    NaN          NaN   
2      Piso                 NaN    NaN       0.0    NaN          NaN   
3      Piso                 NaN    NaN       0.0    NaN          NaN   
4      Piso                 NaN    NaN       0.0    NaN          NaN   

   dormitorios banos  balcon  terraza garaje  trastero  ascensor porteria  
0          3.0     1     NaN      NaN    0.0       1.0       NaN      NaN  
1          4.0     2     NaN      NaN    1.0       0.0       NaN      NaN  
2          5.0     3     NaN      NaN    0.0       0.0       NaN      NaN  
3          1.0     1     NaN      NaN    0.0       0.0       NaN      NaN  
4          3.0     1     NaN      NaN    0.0       0.0       NaN      NaN

In [8]:
#Muestreo de las ultimas 5 filas 
df_131.tail()

title  \
10236   NaN   
10237   NaN   
10238   NaN   
10239   NaN   
10240   NaN   

                                                                                                                                  url  \
10236                              https://www.redpiso.es/inmueble/piso-en-venta-en-calle-narvaez-retiro-madrid-madrid-RP962024124554   
10237       https://www.redpiso.es/inmueble/piso-en-venta-en-calle-fernandez-de-la-hoz-almagro-chamberi-madrid-madrid-RP1922024124579   
10238                                     https://www.redpiso.es/inmueble/piso-en-venta-en-aluche-latina-madrid-madrid-RP272023121667   
10239          https://www.redpiso.es/inmueble/piso-en-venta-en-calle-fondon-apostol-santiago-hortaleza-madrid-madrid-RP2532022104355   
10240  https://www.redpiso.es/inmueble/piso-en-venta-en-calle-infanta-mercedes-11-cuatro-caminos-tetuan-madrid-madrid-RP2332024131076   

          precio  precio_anterior descuento  EUR/m2  m2_constr   distrito  \
10236  1450000.0              NaN       NaN     NaN      158.0     retiro   
10237  1990000.0              NaN       NaN     NaN      317.0   chamberi   
10238   110000.0              NaN       NaN     NaN       68.0     latina   
10239   170000.0              NaN       NaN     NaN      107.0  hortaleza   
10240   420000.0              NaN       NaN     NaN       61.0     tetuan   

       cod_distrito            barrio  cod_barrio                    zona  \
10236           3.0               NaN         NaN                  RETIRO   
10237           7.0           almagro        74.0       CHAMBERÍ  ALMAGRO   
10238          10.0            aluche       104.0                  ALUCHE   
10239          16.0  apostol santiago       165.0               MANOTERAS   
10240           6.0    cuatro caminos        62.0  CUATRO CAMINOSBERNABÉU   

                         calle detalle consumoce_ano letrace emisiones_co2  \
10236    -calle-narvaez-retiro     NaN           NaN     NaN           NaN   
10237      -calle-fernandez-de     NaN           NaN     NaN           NaN   
10238           -aluche-latina     NaN           NaN     NaN           NaN   
10239    -calle-fondon-apostol     NaN           NaN     NaN           NaN   
10240  -calle-infanta-mercedes     NaN           NaN     NaN           NaN   

      tipologia  propiedad_completa                  estado amueblado planta  \
10236      Piso                 NaN  Estado Perfecto Estado       0.0    NaN   
10237      Piso                 NaN       Estado A reformar       0.0    NaN   
10238      Piso                 NaN       Estado A reformar       0.0    NaN   
10239      Piso                 NaN  Estado Perfecto Estado       0.0    NaN   
10240      Piso                 NaN  Estado Perfecto Estado       0.0    NaN   

      habitaciones  dormitorios banos  balcon  terraza     garaje  trastero  \
10236          NaN          3.0   3.0     NaN      NaN  Garajes 1       1.0   
10237          NaN          4.0   3.0     NaN      NaN  Garajes 2       1.0   
10238          NaN          2.0   1.0     NaN      NaN        NaN       0.0   
10239          NaN          3.0   2.0     NaN      NaN        NaN       0.0   
10240          NaN          2.0   1.0     NaN      NaN        NaN       0.0   

       ascensor porteria  
10236       1.0      NaN  
10237       1.0      NaN  
10238       0.0      NaN  
10239       1.0      NaN  
10240       1.0      NaN

In [9]:
# Muestreo de 5 filas aleatorias 
df_131.sample(5)

title  \
8624                  Piso en venta en Ventas   
9496                                      NaN   
6375                    Piso en venta en Goya   
4623  Chalet adosado en venta en Nueva España   
385          Estudio en venta en Ibiza Madrid   

                                                                                                                         url  \
8624                                                      https://www.pisos.com/comprar/piso-ventas28017-45887879128_100200/   
9496  https://www.redpiso.es/inmueble/piso-en-venta-en-plaza-carballo-pilar-fuencarral-el-pardo-madrid-madrid-RP122024122124   
6375                                                        https://www.pisos.com/comprar/piso-goya28009-40071818436_100500/   
4623                                      https://www.pisos.com/comprar/chalet_adosado-nueva_espana28036-41673523991_149200/   
385                                        https://www.pisosmadrid.com.es/propiedad/4004990-estudio-en-venta-en-ibiza-madrid   

         precio  precio_anterior descuento  EUR/m2  m2_constr  \
8624   169000.0              NaN       NaN  2640.0       64.0   
9496   225000.0              NaN       NaN     NaN       56.0   
6375   970000.0              NaN       NaN  6928.0      140.0   
4623  3110000.0              NaN       NaN  7994.0      389.0   
385    320000.0              NaN       NaN     NaN       41.0   

                 distrito  cod_distrito        barrio  cod_barrio  \
8624        Ciudad Lineal          15.0        Ventas       151.0   
9496  Fuencarral-El Pardo           8.0         pilar        84.0   
6375            Salamanca           4.0          Goya        42.0   
4623            Chamartín           5.0  Nueva España        55.0   
385                Retiro           1.0         Ibiza        12.0   

                  zona                  calle  \
8624               NaN                    NaN   
9496  BARRIO DEL PILAR  -plaza-carballo-pilar   
6375               NaN                    NaN   
4623               NaN                    NaN   
385              Ibiza                    NaN   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            detalle

In [10]:
# Chqueo de nulos en todas las columnas
df_131.isna().sum()

title                  1470
url                       0
precio                    0
precio_anterior       10186
descuento             10186
EUR/m2                 2381
m2_constr                28
distrito                177
cod_distrito              9
barrio                  419
cod_barrio               77
zona                   4812
calle                  7800
detalle                4227
consumoce_ano          7091
letrace                8930
emisiones_co2          7091
tipologia                 0
propiedad_completa    10128
estado                 4666
amueblado              7371
planta                 4564
habitaciones           7175
dormitorios             187
banos                   296
balcon                 4383
terraza                4240
garaje                 8201
trastero               1950
ascensor               2222
porteria              10092
dtype: int64

In [11]:
# Muestreo estadistico de las columnas
df_131.describe().T

count           mean           std      min         25%  \
precio              10241.0  944530.017869  1.224786e+06      1.0  248000.000   
precio_anterior        55.0  830694.545455  9.775557e+05  98100.0  201500.000   
EUR/m2               7860.0    3716.766930  4.009314e+03      0.0       6.883   
m2_constr           10213.0     183.331492  6.218287e+02      0.0      67.000   
cod_distrito        10232.0       7.633210  5.718893e+00      0.0       4.000   
cod_barrio          10164.0      79.448249  5.728048e+01      0.0      36.000   
propiedad_completa    113.0       1.000000  0.000000e+00      1.0       1.000   
dormitorios         10054.0       2.814999  1.403123e+00      0.0       2.000   
balcon               5858.0       0.144418  3.515430e-01      0.0       0.000   
terraza              6001.0       0.318780  4.660425e-01      0.0       0.000   
trastero             8291.0       0.155229  3.621443e-01      0.0       0.000   
ascensor             8019.0       0.526500  4.993284e-01      0.0       0.000   

                         50%        75%         max  
precio              515000.0  1150000.0  19950000.0  
precio_anterior     440000.0   955000.0   4900000.0  
EUR/m2                2972.5     6340.5     27708.0  
m2_constr               97.0      150.0     17324.0  
cod_distrito             6.0       11.0        21.0  
cod_barrio              63.0      117.0       215.0  
propiedad_completa       1.0        1.0         1.0  
dormitorios              3.0        3.0        24.0  
balcon                   0.0        0.0         1.0  
terraza                  0.0        1.0         1.0  
trastero                 0.0        0.0         1.0  
ascensor                 1.0        1.0         1.0

In [12]:
# Muestreo de tipos de columnas y nulos 
df_131.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10241 entries, 0 to 10240
Data columns (total 31 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   title               8771 non-null   object 
 1   url                 10241 non-null  object 
 2   precio              10241 non-null  float64
 3   precio_anterior     55 non-null     float64
 4   descuento           55 non-null     object 
 5   EUR/m2              7860 non-null   float64
 6   m2_constr           10213 non-null  float64
 7   distrito            10064 non-null  object 
 8   cod_distrito        10232 non-null  float64
 9   barrio              9822 non-null   object 
 10  cod_barrio          10164 non-null  float64
 11  zona                5429 non-null   object 
 12  calle               2441 non-null   object 
 13  detalle             6014 non-null   object 
 14  consumoce_ano       3150 non-null   object 
 15  letrace             1311 non-null   object 
 16  emis

In [13]:
# Chequeo de nulos en la columna title 
df_131["title"].isna().sum()

1470

In [14]:
# Filtro para checar si las filas que no tienen valores en title tienen valores en otras columnas importantes
filas_con_nulos= df_131[df_131["title"].isnull()]
filas_con_nulos

title  \
8771    NaN   
8772    NaN   
8773    NaN   
8774    NaN   
8775    NaN   
...     ...   
10236   NaN   
10237   NaN   
10238   NaN   
10239   NaN   
10240   NaN   

                                                                                                                                  url  \
8771          https://www.redpiso.es/inmueble/piso-en-venta-en-calle-benimamet-san-cristobal-villaverde-madrid-madrid-RP2842024125161   
8772                                              https://www.redpiso.es/inmueble/piso-en-venta-en-usera-madrid-madrid-RP782024130863   
8773                             https://www.redpiso.es/inmueble/piso-en-venta-en-buenavista-carabanchel-madrid-madrid-RP442024126007   
8774                             https://www.redpiso.es/inmueble/piso-en-venta-en-buenavista-carabanchel-madrid-madrid-RP442024126427   
8775          https://www.redpiso.es/inmueble/piso-en-venta-en-calle-beniferri-san-cristobal-villaverde-madrid-madrid-RP2842024130186   
...                                                                                                                               ...   
10236                              https://www.redpiso.es/inmueble/piso-en-venta-en-calle-narvaez-retiro-madrid-madrid-RP962024124554   
10237       https://www.redpiso.es/inmueble/piso-en-venta-en-calle-fernandez-de-la-hoz-almagro-chamberi-madrid-madrid-RP1922024124579   
10238                                     https://www.redpiso.es/inmueble/piso-en-venta-en-aluche-latina-madrid-madrid-RP272023121667   
10239          https://www.redpiso.es/inmueble/piso-en-venta-en-calle-fondon-apostol-santiago-hortaleza-madrid-madrid-RP2532022104355   
10240  https://www.redpiso.es/inmueble/piso-en-venta-en-calle-infanta-mercedes-11-cuatro-caminos-tetuan-madrid-madrid-RP2332024131076   

          precio  precio_anterior descuento  EUR/m2  m2_constr     distrito  \
8771    106500.0              NaN       NaN     NaN       64.0   villaverde   
8772    119000.0              NaN       NaN     NaN       30.0        usera   
8773    120000.0              NaN       NaN     NaN       37.0  carabanchel   
8774    120000.0              NaN       NaN     NaN       33.0  carabanchel   
8775    127500.0              NaN       NaN     NaN       58.0   villaverde   
...          ...              ...       ...     ...        ...          ...   
10236  1450000.0              NaN       NaN     NaN      158.0       retiro   
10237  1990000.0              NaN       NaN     NaN      317.0     chamberi   
10238   110000.0              NaN       NaN     NaN       68.0       latina   
10239   170000.0              NaN       NaN     NaN      107.0    hortaleza   
10240   420000.0              NaN       NaN     NaN       61.0       tetuan   

       cod_distrito            barrio  cod_barrio                    zona  \
8771           17.0     san cristobal       172.0           SAN CRISTOBAL   
8772           12.0         orcasitas       121.0          USERAORCASITAS   
8773           11.0        buenavista       116.0         PAU CARABANCHEL   
8774           11.0        buenavista       116.0         PAU CARABANCHEL   
8775           17.0     san cristobal       172.0           SAN CRISTOBAL   
...             ...               ...         ...                     ...   
10236           3.0               NaN         NaN                  RETIRO   
10237           7.0           almagro        74.0       CHAMBERÍ  ALMAGRO   
10238          10.0            aluche       104.0                  ALUCHE   
10239          16.0  apostol santiago       165.0               MANOTERAS   
10240           6.0    cuatro caminos        62.0  CUATRO CAMINOSBERNABÉU   

                         calle detalle consumoce_ano letrace emisiones_co2  \
8771      -calle-benimamet-san     NaN           NaN     NaN           NaN   
8772                    -usera     NaN           NaN     NaN           NaN   
8773   -buenavista-carabanchel     NaN           NaN     NaN           N

In [15]:
df_131["title"].head(20)

0                             Piso en venta en Madrid
1     Piso en venta en Montecarmelo  El Goloso Madrid
2                    Piso en venta en Malasana Madrid
3                Piso en venta en Pueblo Nuevo Madrid
4                  Piso en venta en Villaverde Madrid
5             Apartamento en venta en Malasana Madrid
6                Apartamento en venta en Ibiza Madrid
7     Apartamento en venta en Fuente del Berro Madrid
8            Apartamento en venta en Salamanca Madrid
9                   Piso en venta en San Diego Madrid
10                  Piso en venta en Entrevias Madrid
11                            Piso en venta en Madrid
12         Piso en venta en Puente de Vallecas Madrid
13                            Piso en venta en Madrid
14                   Piso en venta en Comillas Madrid
15                     Piso en venta en Centro Madrid
16                    Piso en venta en Aravaca Madrid
17                  Piso en venta en Vicalvaro Madrid
18                          

In [16]:
df_131["title"].sample(20)

6722                                                             Piso en venta en Chueca
6193                                                           Piso en venta en Valverde
4319                                                           Ático en venta en Moncloa
6112        Piso en venta en Calle Arte Figurativo Con Avenida Gran Vía del Sureste, S/N
5555                                                      Piso en venta en Ciudad Jardín
2846                                     Piso de tres habitaciones 167 m², Ibiza, Madrid
8211                                               Piso en venta en Carrera San Jerónimo
2059                                   Piso Plaza De Los Reyes Magos, Niño Jesús, Madrid
4405            Casa en venta en Calle de Méntrida, cerca de Calle del Arroyo del Quinto
2956                                    Piso de tres habitaciones 60 m², Aravaca, Madrid
3722                              Piso de una habitación 73 m², Apóstol Santiago, Madrid
4475                 

In [17]:
df_131["url"].sample(10)

2789                                                                                         https://www.indomio.es/anuncios/96632249/
5347                                           https://www.pisos.com/comprar/piso-centro_embajadores_lavapies28012-42523407114_100200/
6654                                                             https://www.pisos.com/comprar/piso-jeronimos28014-42595073631_526309/
4203                                          https://www.pisos.com/comprar/atico-centro_embajadores_lavapies28012-44199941041_108500/
1127                                                                                         https://www.indomio.es/anuncios/94225921/
1812                                                                                         https://www.indomio.es/anuncios/96074835/
7133                                                          https://www.pisos.com/comprar/piso-montecarmelo28049-45088915150_528410/
6454                                                   

In [18]:
df_131["url"].isna().sum()

0

In [ ]:
# Filtro de filas con valores pequeños que no podrian ser el coste de un inmueble
filas_valor_1 = df_131[df_131["precio"] == 1]
filas_valor_1

title  \
774               Piso en venta en Media Legua Madrid   
1667  Piso de una habitación planta baja, Sol, Madrid   

                                                                                       url  \
774   https://www.pisosmadrid.com.es/propiedad/4068502-piso-en-venta-en-media-legua-madrid   
1667                                             https://www.indomio.es/anuncios/95899949/   

      precio  precio_anterior descuento  EUR/m2  m2_constr   distrito  \
774      1.0              NaN       NaN     NaN      104.0  Moratalaz   
1667     1.0              NaN       NaN     NaN        NaN     Centro   

      cod_distrito       barrio  cod_barrio         zona calle  \
774            1.0  Media Legua        16.0  Media Legua   NaN   
1667           1.0          Sol        16.0          Sol   NaN   

                                 detalle consumoce_ano letrace emisiones_co2  \
774                                  NaN           NaN     NaN           NaN   
1667  Piso de una habitación planta baja   desconocido     NaN   No indicado   

     tipologia  propiedad_completa estado amueblado       planta habitaciones  \
774       Piso                 NaN    NaN       1.0          NaN          NaN   
1667      Piso                 NaN    NaN       NaN  Planta baja            1   

      dormitorios banos  balcon  terraza garaje  trastero  ascensor porteria  
774           3.0     2     NaN      NaN    0.0       1.0       NaN      NaN  
1667          1.0     1     0.0      NaN    NaN       1.0       1.0      NaN

In [20]:
# Filtro de las filas con valores en precio anteior 
filas_sin_nulos= df_131[~df_131["precio_anterior"].isnull()]
filas_sin_nulos.head(4)

title  \
1390  Piso de cuatro habitaciones Calle Puerto Del Bruch, Santa Eugenia, Madrid   
1671                Piso de tres habitaciones Calle Sorbe, 17, Comillas, Madrid   
1726                                             Dúplex 141 m², Piovera, Madrid   
1738              Piso de tres habitaciones 87 m², Embajadores-Lavapiés, Madrid   

                                            url    precio  precio_anterior  \
1390  https://www.indomio.es/anuncios/95400629/  132000.0         139000.0   
1671  https://www.indomio.es/anuncios/95918757/  121000.0         131100.0   
1726  https://www.indomio.es/anuncios/95942167/  790000.0         870000.0   
1738  https://www.indomio.es/anuncios/95954689/  549000.0         585000.0   

     descuento  EUR/m2  m2_constr            distrito  cod_distrito  \
1390     -5,0%   2.237       59.0   Villa de Vallecas          18.0   
1671     -7,7%   1.592       76.0         Carabanchel          11.0   
1726     -9,2%   5.603      141.0           Hortaleza          16.0   
1738     -6,2%   6.310       87.0              Centro           1.0   

             barrio  cod_barrio           zona                   calle  \
1390  Santa Eugenia       182.0  Santa Eugenia  calle puerto del bruch   
1671       Comillas       111.0       Comillas         calle sorbe, 17   
1726        Piovera       162.0        Piovera                     NaN   
1738    Embajadores        12.0    Embajadores                     NaN   

                                                 detalle consumoce_ano  \
1390  Piso de cuatro habitaciones Calle Puerto Del Bruch   desconocido   
1671           Piso de tres habitaciones Calle Sorbe, 17   desconocido   
1726                                       Dúplex 141 m²   desconocido   
1738                     Piso de tres habitaciones 87 m²   desconocido   

     letrace   emisiones_co2 tipologia  propiedad_completa         estado  \
1390     NaN     No indicado      Piso                 NaN            NaN   
1671     NaN  Inclasificable      Piso                 NaN  Para reformar   
1726     NaN     No indicado    Dúplex                 NaN            NaN   
1738     NaN     No indicado      Piso                 NaN            NaN   

     amueblado planta habitaciones  dormitorios banos  balcon  terraza garaje  \
1390       NaN      3            4          4.0     1     NaN      NaN    NaN   
1671       NaN      2            3          3.0     1     0.0      NaN    NaN   
1726       NaN    NaN            3          3.0     2     NaN      NaN    NaN   
1738       NaN    NaN            3          3.0     1     NaN      NaN    NaN   

      trastero  ascensor porteria  
1390       NaN       NaN      NaN  
1671       NaN       0.0      NaN  
1726       0.0       NaN      NaN  
1738       0.0       NaN      NaN

In [21]:
# Aqui filtre los links de los inmuebles con el precio anterior para checar algunos directamente y 
# ver si habia algo relevante del porque del cambio del precio.
filas_sin_nulos["url"]

1390    https://www.indomio.es/anuncios/95400629/
1671    https://www.indomio.es/anuncios/95918757/
1726    https://www.indomio.es/anuncios/95942167/
1738    https://www.indomio.es/anuncios/95954689/
1747    https://www.indomio.es/anuncios/95961129/
1754    https://www.indomio.es/anuncios/95978631/
1799    https://www.indomio.es/anuncios/96067207/
1866    https://www.indomio.es/anuncios/96144787/
1886    https://www.indomio.es/anuncios/96164955/
1887    https://www.indomio.es/anuncios/96164957/
1888    https://www.indomio.es/anuncios/96164967/
1902    https://www.indomio.es/anuncios/96180671/
1931    https://www.indomio.es/anuncios/96208553/
1939    https://www.indomio.es/anuncios/96233757/
1983    https://www.indomio.es/anuncios/96295235/
2071    https://www.indomio.es/anuncios/96361185/
2080    https://www.indomio.es/anuncios/96364727/
2081    https://www.indomio.es/anuncios/96364731/
2093    https://www.indomio.es/anuncios/96365933/
2115    https://www.indomio.es/anuncios/96371557/


In [22]:
# Chequeo de porcentaje de nulos en la columna precio_anterior
df_131["precio_anterior"].isna().mean()*100

99.46294307196563

In [23]:
df_131.shape

(10241, 31)

In [24]:
# Porcentaje de nulos 
df_131["descuento"].isna().mean()*100

99.46294307196563

In [25]:
#Porcentaje de nulos
df_131["EUR/m2"].isna().mean()*100

23.249682648178887

In [26]:
# Chequeo de nulos totales
df_131["EUR/m2"].isna().sum()

2381

In [27]:
df_131["EUR/m2"].describe()

count     7860.000000
mean      3716.766930
std       4009.313963
min          0.000000
25%          6.883000
50%       2972.500000
75%       6340.500000
max      27708.000000
Name: EUR/m2, dtype: float64

In [28]:
# Filtro de valores iguales a 0
filas_valor_eur= df_131[df_131["EUR/m2"] == 0]
filas_valor_eur

title  \
1949  Piso 656600 m², Recoletos, Madrid   

                                            url    precio  precio_anterior  \
1949  https://www.indomio.es/anuncios/96246263/  124150.0              NaN   

     descuento  EUR/m2  m2_constr    distrito  cod_distrito     barrio  \
1949       NaN     0.0      656.6   Salamanca           4.0  Recoletos   

      cod_barrio       zona calle         detalle consumoce_ano letrace  \
1949        41.0  Recoletos   NaN  Piso 656600 m²   desconocido       G   

     emisiones_co2 tipologia  propiedad_completa estado amueblado planta  \
1949   No indicado      Piso                 NaN    NaN       NaN    NaN   

     habitaciones  dormitorios banos  balcon  terraza garaje  trastero  \
1949          NaN          NaN   NaN     NaN      NaN    NaN       0.0   

      ascensor porteria  
1949       NaN      NaN

In [29]:
# Filtro de valores 
filas_valor_eur_pequeños= df_131[df_131["EUR/m2"] <=100 ]
filas_valor_eur_pequeños.tail(5)

title  \
8653  Piso en venta en Calle del Abad Juan Catalán, s/n   
8737             Piso en venta en Calle Arroyo Bueno, 1   
8739             Piso en venta en Calle Arroyo Bueno, 1   
8745      Piso en venta en Plaza Mayor de Villaverde, 8   
8749             Piso en venta en Calle Arroyo Bueno, 1   

                                                                                   url  \
8653       https://www.pisos.com/comprar/piso-vicalvaro_ambroz28032-4827661777_109700/   
8737  https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4751773116_109700/   
8739  https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4782740391_109700/   
8745  https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4885463236_109700/   
8749  https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4918181785_109700/   

        precio  precio_anterior descuento  EUR/m2  m2_constr    distrito  \
8653  465000.0              NaN       NaN     2.0        2.0   Vicálvaro   
8737  242300.0              NaN       NaN     1.0        2.0  Villaverde   
8739  348500.0              NaN       NaN     4.0        2.0  Villaverde   
8745  348000.0              NaN       NaN     1.0        2.0  Villaverde   
8749  277700.0              NaN       NaN     6.0        2.0  Villaverde   

      cod_distrito                                     barrio  cod_barrio  \
8653          19.0                  CascoHistóricodeVicálvaro       191.0   
8737          17.0  VillaverdeAlto,CascoHistóricodeVillaverde       171.0   
8739          17.0  VillaverdeAlto,CascoHistóricodeVillaverde       171.0   
8745          17.0  VillaverdeAlto,CascoHistóricodeVillaverde       171.0   
8749          17.0  VillaverdeAlto,CascoHistóricodeVillaverde       171.0   

     zona calle  \
8653  bar   NaN   
8737  NaN   NaN   
8739  NaN   NaN   
8745  NaN   NaN   
8749  NaN   NaN   

                                                                                                                                                                                                                                                                      detalle  \
8653                                           2B. Piso en un inmueble de obra nueva de 148 m2 de superficie construida y situado en la zona de Madrid, acomodado en planta 2. Está estructurado en 3 habitaciones, 2 baños completos y terraza. Aporta como extras ascensor.   
8737                  9.2D. Piso en un inmueble de obra nueva de 77 m2 de superficie construida y situado en la zona de Villaverde.  Acomodado en planta 1. Está estructurado en 2 habitaciones, 2 baños completos y terraza. Aporta como extras ascensor, garaje y trastero.   
8739                 3.3D. Piso en un inmueble de obra nueva de 104 m2 de superficie construida y situado en la zona de Villaverde.  Acomodado en planta 4. Está estructurado en 3 habitaciones, 2 baños completos y terraza. Aporta como extras ascensor, garaje y trastero.   
8745  VA40000199301. Piso en un edificio de obra nueva de 134 m2 de superficie construida y ubicado en la zona de Madrid, enclavado en planta 1. Está distribuido en 3 habitaciones, 2 baños completos y terraza de 6.30 m2. Incluye extras como ascensor, garaje y trastero.   
8749                 10.2D. Piso en un inmueble de obra nueva de 88 m2 de superficie construida y situado en la zona de Villaverde.  Acomodado en planta 1. Está estructurado en 2 habitaciones, 2 baños completos y terraza. Aporta como extras ascensor, garaje y trastero.   

     consumoce_ano letrace emisiones_co2 tipologia  propiedad_completa  \
8653           NaN     NaN           NaN      Piso                 NaN   
8737           NaN     NaN           NaN      Piso                 NaN   
8739           NaN     NaN           NaN      Piso                 NaN   
8745           NaN     NaN           NaN      Piso                 NaN   
8749           NaN     NaN           NaN      Piso                 NaN   

          estado amueblado planta h

In [30]:
# Filtro para valores menores a 99 con un decimal. de esa forma los valores podran ser filtrados para no tener un precio mayor a 4 cifras por metro cuadrado
# que es lo que se ha notado donde predominan los valores.  Aqui cheque los valores con Data Wrangler y pude visualizar que los valores no son correctos. A algunos 
# les sirve solo colocarle 000 ceros extras al valor por metro cuadrado pero a otros no les sirve eso. 

filtered_df_131 = df_131[df_131['EUR/m2'].apply(lambda x: x < 99 and (x * 10).is_integer())]
filtered_df_131.head(5)

title  \
893                                Ático Calle de Núñez de Balboa 2, Recoletos, Madrid   
898  Piso de dos habitaciones Avenida de Alberto de Alcocer 12, Hispanoamérica, Madrid   
904             Piso de tres habitaciones buen estado, planta baja, San Andrés, Madrid   
912                            Piso de dos habitaciones buen estado, San Diego, Madrid   
928                    Piso de una habitación Calle Juan de Austria, Trafalgar, Madrid   

                                           url     precio  precio_anterior  \
893  https://www.indomio.es/anuncios/75404850/  2000000.0              NaN   
898  https://www.indomio.es/anuncios/82164753/   599999.0              NaN   
904  https://www.indomio.es/anuncios/83875795/   165000.0              NaN   
912  https://www.indomio.es/anuncios/88009171/   210000.0              NaN   
928  https://www.indomio.es/anuncios/88772857/   225000.0              NaN   

    descuento  EUR/m2  m2_constr             distrito  cod_distrito  \
893       NaN    20.0      100.0            Salamanca           4.0   
898       NaN     7.5       80.0            Chamartín           5.0   
904       NaN     2.2       75.0                  NaN           0.0   
912       NaN     2.8       75.0   Puente de Vallecas          13.0   
928       NaN     4.5       50.0             Chamberí           7.0   

             barrio  cod_barrio            zona  \
893       Recoletos        41.0       Recoletos   
898  Hispanoamérica        54.0  Hispanoamérica   
904             NaN         0.0      San Andrés   
912       San Diego       132.0       San Diego   
928       Trafalgar        73.0       Trafalgar   

                                calle  \
893        calle de nunez de balboa 2   
898  avenida de alberto de alcocer 12   
904                               NaN   
912                               NaN   
928             calle juan de austria   

                                                       detalle consumoce_ano  \
893                           Ático Calle de Núñez de Balboa 2        ≥ 3,51   
898  Piso de dos habitaciones Avenida de Alberto de Alcocer 12   desconocido   
904         Piso de tres habitaciones buen estado, planta baja        ≥ 3,51   
912                       Piso de dos habitaciones buen estado   desconocido   
928               Piso de una habitación Calle Juan de Austria   desconocido   

    letrace emisiones_co2 tipologia  propiedad_completa             estado  \
893       F             D     Ático                 1.0      Para reformar   
898     NaN   No indicado      Piso                 NaN                NaN   
904       F     Pendiente      Piso                 1.0  Bueno / Habitable   
912       B   No indicado      Piso                 NaN  Bueno / Habitable   
928     NaN   No indicado      Piso                 NaN  Bueno / Habitable   

                 amueblado planta habitaciones  dormitorios banos  balcon  \
893  Sólo cocina amueblada      1            1          1.0     1     1.0   
898                    NaN      8            2          2.0     1     NaN   
904                     Sí      1            3          3.0     1     1.0   
912                     No      1            2          2.0     1     1.0   
928                     Sí      4            1          1.0     1     NaN   

     terraza               garaje  trastero  ascensor               porteria  
893      1.0  2 en garaje privado       1.0       1.0  Portero media jornada  
898      NaN                  NaN       0.0       1.0                    NaN  
904      1.0                  NaN       1.0       0.0                    NaN  
912      NaN                  NaN       1.0       NaN                    NaN  
928      NaN                  NaN       0.0       0.0                    NaN

In [31]:
# Filtro por valores que parecen atipicos e irregulares para identificar vericidad
filas_valor_eur2= df_131[df_131["EUR/m2"] >= 20000]
filas_valor_eur2.head()

title  \
4310  Ático en venta en Calle de José Ortega y Gasset, cerca de Calle del Conde de Peñalver   
4332                    Ático en venta en Paseo de la Habana, cerca de Calle de la Macarena   
4349               Ático en venta en Calle de Núñez de Balboa, cerca de Calle de Villanueva   
4355                                     Ático en venta en Barrio de Salamanca - Castellana   
6652                                                  Piso en venta en Calle de Alfonso XII   

                                                                                    url  \
4310          https://www.pisos.com/comprar/atico-lista_barrio28006-44241474297_101800/   
4332          https://www.pisos.com/comprar/atico-nueva_espana28036-33358182214_996542/   
4349             https://www.pisos.com/comprar/atico-recoletos28001-47515636484_108900/   
4355  https://www.pisos.com/comprar/atico-salamanca_castellana28006-48379619739_101800/   
6652              https://www.pisos.com/comprar/piso-jeronimos28014-40847627180_996542/   

          precio  precio_anterior descuento   EUR/m2  m2_constr   distrito  \
4310   2490000.0              NaN       NaN  21652.0      115.0  Salamanca   
4332   4625000.0              NaN       NaN  22671.0      204.0  Chamartín   
4349   3950000.0              NaN       NaN  20466.0      193.0  Salamanca   
4355  19950000.0              NaN       NaN  27708.0      720.0  Salamanca   
6652   5500000.0              NaN       NaN  20833.0      264.0     Retiro   

      cod_distrito        barrio  cod_barrio zona calle  \
4310           4.0         Lista        45.0  NaN   NaN   
4332           5.0  Nueva España        55.0  NaN   NaN   
4349           4.0     Recoletos        41.0  NaN   NaN   
4355           4.0    Castellana        46.0  NaN   NaN   
6652           3.0     Jerónimos        35.0  NaN   NaN   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                

In [32]:
# Filtro del link por valores que parecen atipicos e irregulares para identificar vericidad
filas_valor_eur2= df_131[df_131["EUR/m2"] >= 20000]
filas_valor_eur2["url"]

4310            https://www.pisos.com/comprar/atico-lista_barrio28006-44241474297_101800/
4332            https://www.pisos.com/comprar/atico-nueva_espana28036-33358182214_996542/
4349               https://www.pisos.com/comprar/atico-recoletos28001-47515636484_108900/
4355    https://www.pisos.com/comprar/atico-salamanca_castellana28006-48379619739_101800/
6652                https://www.pisos.com/comprar/piso-jeronimos28014-40847627180_996542/
6679          https://www.pisos.com/comprar/piso-justicia_chueca28001-44243093919_102100/
6949             https://www.pisos.com/comprar/piso-lista_barrio28006-44251095150_101800/
7799     https://www.pisos.com/comprar/piso-salamanca_castellana28006-34232013494_108900/
Name: url, dtype: object

In [33]:
# Porcentaje de nulos por columna
df_131.isna().mean()*100

title                 14.354067
url                    0.000000
precio                 0.000000
precio_anterior       99.462943
descuento             99.462943
EUR/m2                23.249683
m2_constr              0.273411
distrito               1.728347
cod_distrito           0.087882
barrio                 4.091397
cod_barrio             0.751880
zona                  46.987599
calle                 76.164437
detalle               41.275266
consumoce_ano         69.241285
letrace               87.198516
emisiones_co2         69.241285
tipologia              0.000000
propiedad_completa    98.896592
estado                45.561957
amueblado             71.975393
planta                44.565960
habitaciones          70.061517
dormitorios            1.825994
banos                  2.890343
balcon                42.798555
terraza               41.402207
garaje                80.080070
trastero              19.041109
ascensor              21.697100
porteria              98.545064
dtype: f

In [34]:
# Analisis descriptivo de la columna m2/constr 
df_131["m2_constr"].describe()

count    10213.000000
mean       183.331492
std        621.828720
min          0.000000
25%         67.000000
50%         97.000000
75%        150.000000
max      17324.000000
Name: m2_constr, dtype: float64

In [35]:
#Filtro de valores menores a 20 metros cuadrados. Se analizo con 10 tambien y siguen siendo mas de 200
filas_valor_m2= df_131[df_131["m2_constr"] <= 10]
filas_valor_m2.shape

(225, 31)

In [36]:
# filtro por columnas url 
filas_valor_m2["url"]

123          https://www.pisosmadrid.com.es/propiedad/3912876-piso-en-venta-en-malasana-madrid
635      https://www.pisosmadrid.com.es/propiedad/4056165-piso-en-venta-en-pueblo-nuevo-madrid
639     https://www.pisosmadrid.com.es/propiedad/4056174-apartamento-en-venta-en-centro-madrid
644     https://www.pisosmadrid.com.es/propiedad/4056185-apartamento-en-venta-en-ventas-madrid
977                                                  https://www.indomio.es/anuncios/91884849/
                                                 ...                                          
8745          https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4885463236_109700/
8746          https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4888683269_109700/
8747          https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4891828236_109700/
8748          https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4915060649_109700/
8749          https://www.pisos.com/comprar/piso-v

In [37]:
# Muestreo de dos columnas especificas
filas_valor_m2[["url", "m2_constr"]]

url  \
123        https://www.pisosmadrid.com.es/propiedad/3912876-piso-en-venta-en-malasana-madrid   
635    https://www.pisosmadrid.com.es/propiedad/4056165-piso-en-venta-en-pueblo-nuevo-madrid   
639   https://www.pisosmadrid.com.es/propiedad/4056174-apartamento-en-venta-en-centro-madrid   
644   https://www.pisosmadrid.com.es/propiedad/4056185-apartamento-en-venta-en-ventas-madrid   
977                                                https://www.indomio.es/anuncios/91884849/   
...                                                                                      ...   
8745        https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4885463236_109700/   
8746        https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4888683269_109700/   
8747        https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4891828236_109700/   
8748        https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4915060649_109700/   
8749        https://www.pisos.com/comprar/piso-villaverde_san_andres28021-4918181785_109700/   

      m2_constr  
123       0.000  
635       0.000  
639       0.000  
644       0.000  
977       1.775  
...         ...  
8745      2.000  
8746      1.000  
8747      1.000  
8748      2.000  
8749      2.000  

[225 rows x 2 columns]

In [38]:
#Muestreo/Filtro de las filas sin valores nulos 
m2_constr_filtro= df_131[~df_131["m2_constr"].isna()]
m2_constr_filtro.sample(5)                                                                                                                          

title  \
1169  Piso de dos habitaciones cuarta planta, Universidad-Malasaña, Madrid   
4301                                    Ático en venta en Calle del Clavel   
6103                                 Piso en venta en Ensanche de Vallecas   
651                   Apartamento planta baja en venta en Salamanca Madrid   
2860     Piso de una habitación Calle De San Pedro, Cortes-Huertas, Madrid   

                                                                                                        url  \
1169                                                              https://www.indomio.es/anuncios/94456785/   
4301                           https://www.pisos.com/comprar/atico-justicia_chueca28004-48417661471_101800/   
6103           https://www.pisos.com/comprar/piso-ensanche_de_vallecas_valdecarros28031-47598701879_101000/   
651   https://www.pisosmadrid.com.es/propiedad/4057888-apartamento-planta-baja-en-venta-en-salamanca-madrid   
2860                                                              https://www.indomio.es/anuncios/96649783/   

        precio  precio_anterior descuento     EUR/m2  m2_constr  \
1169  799000.0              NaN       NaN      8.237       97.0   
4301  995000.0              NaN       NaN  13266.000       75.0   
6103  249000.0              NaN       NaN   3557.000       70.0   
651   325000.0              NaN       NaN        NaN       95.0   
2860  349000.0         369000.0     -5,4%      7.756       45.0   

               distrito  cod_distrito                barrio  cod_barrio  \
1169             Centro           1.0           Universidad        15.0   
4301             Centro           1.0              Justicia        14.0   
6103  Villa de Vallecas          18.0  Ensanche de Vallecas       183.0   
651           Salamanca           1.0                   NaN        12.0   
2860             Centro           1.0                Cortes        13.0   

             zona               calle  \
1169  Universidad                 NaN   
4301          NaN                 NaN   
6103          NaN                 NaN   
651     Salamanca                 NaN   
2860       Cortes  calle de san pedro   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        

In [39]:
# Alcance a terminar m2_constr y me quede en distrito.
df_131["distrito"].unique()

array([nan, 'Fuencarral-El Pardo', 'centro', 'Ciudad Lineal',
       'Villaverde', 'Centro', 'Retiro', 'Salamanca',
       'Puente de Vallecas', 'Carabanchel', 'Moncloa-Aravaca',
       'Arganzuela', 'Latina', 'Moratalaz', 'vicalvaro', 'Tetuán',
       'San Blas-Canillejas', 'Barajas', 'Chamberí', 'Usera', 'Chamartín',
       'Hortaleza', 'tetuan', 'chamartin', 'chamberi', 'ciudad lineal',
       'villaverde', 'Villa de Vallecas', ' Salamanca', ' Barajas',
       ' Arganzuela', ' Chamartín', ' Tetuán', ' Chamberí', ' Latina',
       ' Centro', ' Fuencarral-El Pardo', ' Villa de Vallecas',
       ' Puente de Vallecas', ' Retiro', ' San Blas-Canillejas',
       ' Ciudad Lineal', ' Moncloa-Aravaca', ' Carabanchel', ' Hortaleza',
       ' Vicálvaro', ' Usera', ' Moratalaz', ' Villaverde', 'Vicálvaro',
       'usera', 'carabanchel', 'arganzuela', 'puente de vallecas',
       'latina', 'moratalaz', 'hortaleza', 'retiro', ' vicalvaro',
       'salamanca', ' villa de vallecas', ' barajas'], dt

In [40]:
# Muestreo de valores unicos 
df_131["cod_distrito"].unique()

array([ 4., 21.,  2.,  5.,  6.,  7.,  0., 10.,  1.,  8., 18., 13.,  3.,
       20., 15.,  9., 11., 16., 19., 12., 14., 17., nan])

In [41]:
# Muestreo de valores estadisticos 
df_131["cod_distrito"].describe()

count    10232.000000
mean         7.633210
std          5.718893
min          0.000000
25%          4.000000
50%          6.000000
75%         11.000000
max         21.000000
Name: cod_distrito, dtype: float64

In [42]:
# filtro para encontrar los valores iguales a 0 
cero_distrito= df_131[df_131["cod_distrito"]== 0]
cero_distrito

title  \
11                                                                Piso en venta en Madrid   
53                                                         Piso en venta en Centro Madrid   
70                                                                Piso en venta en Madrid   
94                                                       Piso en venta en Malasana Madrid   
139                                                      Piso en venta en Lavapies Madrid   
...                                                                                   ...   
3855                       Piso de tres habitaciones tercera planta, Montecarmelo, Madrid   
3861                  Piso de cuatro habitaciones 110 m², Valdebebas-Valdefuentes, Madrid   
3890             Piso de dos habitaciones buen estado, primera planta, San Andrés, Madrid   
4015                                Piso de dos habitaciones 100 m², Montecarmelo, Madrid   
4033  Piso de tres habitaciones Avenida Del Monasterio De Silos, 29, Montecarmelo, Madrid   

                                                                                    url  \
11             https://www.pisosmadrid.com.es/propiedad/3867131-piso-en-venta-en-madrid   
53      https://www.pisosmadrid.com.es/propiedad/3887270-piso-en-venta-en-centro-madrid   
70             https://www.pisosmadrid.com.es/propiedad/3891215-piso-en-venta-en-madrid   
94    https://www.pisosmadrid.com.es/propiedad/3897237-piso-en-venta-en-malasana-madrid   
139   https://www.pisosmadrid.com.es/propiedad/3919123-piso-en-venta-en-lavapies-madrid   
...                                                                                 ...   
3855                                          https://www.indomio.es/anuncios/96883479/   
3861                                          https://www.indomio.es/anuncios/96884457/   
3890                                          https://www.indomio.es/anuncios/96892019/   
4015                                          https://www.indomio.es/anuncios/96914777/   
4033                                          https://www.indomio.es/anuncios/96921707/   

        precio  precio_anterior descuento  EUR/m2  m2_constr  \
11    165000.0              NaN       NaN     NaN       60.0   
53    425000.0              NaN       NaN     NaN      152.0   
70    899000.0              NaN       NaN     NaN     5275.0   
94    199000.0              NaN       NaN     NaN      118.0   
139   305000.0              NaN       NaN     NaN       80.0   
...        ...              ...       ...     ...        ...   
3855  780000.0              NaN       NaN   5.306      147.0   
3861  795000.0              NaN       NaN   7.227      110.0   
3890  169000.0              NaN       NaN   2.139       79.0   
4015  520000.0              NaN       NaN   5.200      100.0   
4033  625900.0              NaN       NaN   4.347      144.0   

                 distrito  cod_distrito    barrio  cod_barrio          zona  \
11                    NaN           0.0       NaN         0.0           NaN   
53                 Centro           0.0       NaN         0.0        Centro   
70    San Blas-Canillejas           0.0   Amposta         0.0           NaN   
94                 Centro           0.0  Malasaña         0.0      Malasana   
139                Centro           0.0       Sol         0.0      Lavapies   
...                   ...           ...       ...         ...           ...   
3855                  NaN           0.0       NaN         0.0  Montecarmelo   
3861                  NaN           0.0       NaN         0.0    Valdebebas   
3890                  NaN           0.0       NaN         0.0    San Andrés   
4015                  NaN           0.0       NaN         0.0  Montecarmelo   
4033                  NaN           0.0       NaN         0.0  Montecarmelo   

                                    calle  \
11                                    NaN   
53                                    NaN   
70                   

In [43]:
# Muestreo de valores unicos 
df_131["barrio"].unique()

array([nan, 'El Goloso', 'Pueblo Nuevo', 'Malasaña', 'Ibiza',
       'Fuente del Berro', 'lista', 'San Diego', 'entrevias', 'sol',
       'Comillas', 'Embajadores', 'Aravaca', 'Vista Alegre', 'penagrande',
       'Sol', 'Delicias', 'Guindalera', 'Atocha', 'cortes', 'Recoletos',
       'Goya', 'Legazpi', 'Palomeras Sureste', 'los angeles', 'Pavones',
       'Portazgo', 'Buenavista', 'CuatroCaminos', 'Amposta', 'guindalera',
       'Corralejos', 'Almenara', 'Arapiles', 'Los Rosales', 'Quintana',
       'Lucero', 'Arcos', 'san andres', 'Orcasitas', 'Canillejas',
       'San Isidro', 'zofio', 'Palos de Moguer', 'Vallehermoso',
       'Castilla', 'jeronimos', 'Castellana', 'Valdeacederas',
       'Valdefuentes', 'Castillejos', 'Rosas', 'Piovera',
       'hispanoamerica', 'Fuentelarreina', 'Bellas Vistas',
       'Ciudad Universitaria', 'moscardo', 'Pilar', 'concepcion',
       'Prosperidad', 'Abrantes', 'Puerta Bonita', 'Aluche', 'Orcasur',
       'arguelles', 'Gaztambide', 'san fermin', 'M

In [44]:
# Muestreo de valores estadisticos
 df_131["cod_barrio"].describe()

count    10164.000000
mean        79.448249
std         57.280477
min          0.000000
25%         36.000000
50%         63.000000
75%        117.000000
max        215.000000
Name: cod_barrio, dtype: float64

In [ ]:
# Filtro de valores igual a 0
cero_barrio= df_131[df_131["cod_barrio"]== 0]
cero_barrio

title  \
11                                                                Piso en venta en Madrid   
53                                                         Piso en venta en Centro Madrid   
70                                                                Piso en venta en Madrid   
94                                                       Piso en venta en Malasana Madrid   
139                                                      Piso en venta en Lavapies Madrid   
...                                                                                   ...   
3855                       Piso de tres habitaciones tercera planta, Montecarmelo, Madrid   
3861                  Piso de cuatro habitaciones 110 m², Valdebebas-Valdefuentes, Madrid   
3890             Piso de dos habitaciones buen estado, primera planta, San Andrés, Madrid   
4015                                Piso de dos habitaciones 100 m², Montecarmelo, Madrid   
4033  Piso de tres habitaciones Avenida Del Monasterio De Silos, 29, Montecarmelo, Madrid   

                                                                                    url  \
11             https://www.pisosmadrid.com.es/propiedad/3867131-piso-en-venta-en-madrid   
53      https://www.pisosmadrid.com.es/propiedad/3887270-piso-en-venta-en-centro-madrid   
70             https://www.pisosmadrid.com.es/propiedad/3891215-piso-en-venta-en-madrid   
94    https://www.pisosmadrid.com.es/propiedad/3897237-piso-en-venta-en-malasana-madrid   
139   https://www.pisosmadrid.com.es/propiedad/3919123-piso-en-venta-en-lavapies-madrid   
...                                                                                 ...   
3855                                          https://www.indomio.es/anuncios/96883479/   
3861                                          https://www.indomio.es/anuncios/96884457/   
3890                                          https://www.indomio.es/anuncios/96892019/   
4015                                          https://www.indomio.es/anuncios/96914777/   
4033                                          https://www.indomio.es/anuncios/96921707/   

        precio  precio_anterior descuento  EUR/m2  m2_constr  \
11    165000.0              NaN       NaN     NaN       60.0   
53    425000.0              NaN       NaN     NaN      152.0   
70    899000.0              NaN       NaN     NaN     5275.0   
94    199000.0              NaN       NaN     NaN      118.0   
139   305000.0              NaN       NaN     NaN       80.0   
...        ...              ...       ...     ...        ...   
3855  780000.0              NaN       NaN   5.306      147.0   
3861  795000.0              NaN       NaN   7.227      110.0   
3890  169000.0              NaN       NaN   2.139       79.0   
4015  520000.0              NaN       NaN   5.200      100.0   
4033  625900.0              NaN       NaN   4.347      144.0   

                 distrito  cod_distrito    barrio  cod_barrio          zona  \
11                    NaN           0.0       NaN         0.0           NaN   
53                 Centro           0.0       NaN         0.0        Centro   
70    San Blas-Canillejas           0.0   Amposta         0.0           NaN   
94                 Centro           0.0  Malasaña         0.0      Malasana   
139                Centro           0.0       Sol         0.0      Lavapies   
...                   ...           ...       ...         ...           ...   
3855                  NaN           0.0       NaN         0.0  Montecarmelo   
3861                  NaN           0.0       NaN         0.0    Valdebebas   
3890                  NaN           0.0       NaN         0.0    San Andrés   
4015                  NaN           0.0       NaN         0.0  Montecarmelo   
4033                  NaN           0.0       NaN         0.0  Montecarmelo   

                                    calle  \
11                                    NaN   
53                                    NaN   
70                   

In [46]:
# Muestreo de valores unicos 
df_131["zona"].unique()

array([nan, 'Montecarmelo  El Goloso', 'Malasana', 'Pueblo Nuevo',
       'Villaverde', 'Ibiza', 'Fuente del Berro', 'Salamanca',
       'San Diego', 'Entrevias', 'Puente de Vallecas', 'Comillas',
       'Centro', 'Aravaca', 'Vicalvaro', 'Usera', 'Carabanchel',
       'Vista Alegre', 'Penagrande', 'La Latina', 'Guindalera', 'Atocha',
       'Recoletos', 'Goya', 'Legazpi', 'Palomeras Sureste',
       'Ciudad de los Angeles', 'Pavones', 'Villaverde Bajo', 'Sol',
       'Ciudad Lineal', 'Portazgo', 'Tetuan', 'buenavista', 'Moratalaz',
       'cuatrocaminos', 'corralejos', 'almenara', 'Arapiles', 'Vallecas',
       'Los Rosales', 'Los Angeles', 'Quintana', 'Lucero', 'arcos',
       'San Andres', 'Orcasitas', 'Canillejas', 'Embajadores',
       'San Isidro', 'Zofio', 'Palos de Moguer', 'Lavapies',
       'Vallehermoso', 'Valdeacederas', 'valdefuentes', 'aravaca',
       'Castillejos', 'Rios Rosas', 'Piovera', 'Hispanoamerica',
       'Fuentelarreina', 'Bellas Vistas', 'Castilla', 'Arganzuel

In [47]:
# Muestreo de valores unicos 
df_131["calle"].unique()

array([nan, 'calle de nunez de balboa 2', 'plaza de italia', ...,
       '-lista-salamanca', '-calle-angela-gonzalez',
       '-calle-fondon-apostol'], dtype=object)

In [48]:
# Muestreo de valores aleatorios 
df_131["detalle"].sample(15)

2404                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    

In [49]:
# Muestreo de valores aleatorios
df_131["detalle"].sample(15)

6532                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    

In [50]:
# Muestreo de valores totales por cada valor unico
df_131["consumoce_ano"].value_counts()

consumoce_ano
desconocido    2922
≥ 3,51           58
999              20
123              11
400               5
               ... 
2                 1
222               1
163               1
164               1
149               1
Name: count, Length: 109, dtype: int64

In [78]:
#  Muestreo de nulos totales 
df_131["consumoce_ano"].isna().sum()

7091

In [51]:
# Muestreo de los valores unicos 
df_131 ["letrace"].unique()

array([nan, 'F', 'E', 'G', 'D', 'B', 'C', 'A'], dtype=object)

In [52]:
# Muestreo de los valores unicos
df_131["emisiones_co2"].unique()

array([nan, 'D', 'No indicado', 'Pendiente', 'Exento', 'C', 'F',
       '123 kg CO₂/m² añoB', 'E', '53.5 kg CO₂/m² añoE', 'Inclasificable',
       '103 kg CO₂/m² añoG', '38 kg CO₂/m² añoE', '22 kg CO₂/m² añoD',
       '92 kg CO₂/m² añoA', '116 kg CO₂/m² añoE', '6.81 kg CO₂/m² añoB',
       '48.3 kg CO₂/m² añoE', 'G', '180 kg CO₂/m² añoE',
       '37.9 kg CO₂/m² añoE', '366 kg CO₂/m² añoE', '52.7 kg CO₂/m² añoE',
       '54 kg CO₂/m² añoE', '74 kg CO₂/m² añoE', 'B',
       '25.3 kg CO₂/m² añoD', '65.4 kg CO₂/m² añoE', '42 kg CO₂/m² añoE',
       '48 kg CO₂/m² añoE', '47 kg CO₂/m² añoE', '97 kg CO₂/m² añoG',
       '57 kg CO₂/m² añoD', '29 kg CO₂/m² añoE', '35 kg CO₂/m² añoE',
       '63 kg CO₂/m² añoF', '333 kg CO₂/m² añoE', '46 kg CO₂/m² añoE',
       '40 kg CO₂/m² añoE', '58 kg CO₂/m² añoE', '43 kg CO₂/m² añoE',
       '39 kg CO₂/m² añoE', '63.3 kg CO₂/m² añoE', '66.8 kg CO₂/m² añoF',
       '53 kg CO₂/m² añoE', '17 kg CO₂/m² añoD', '67.1 kg CO₂/m² añoF',
       '67 kg CO₂/m² añoF', '

In [53]:
# Muestreo aleatorios de valores en dos columnas 
df_131[["letrace","emisiones_co2"]].sample(10)

letrace emisiones_co2
63       NaN           NaN
577      NaN           NaN
1398     NaN   No indicado
4005       E   No indicado
7112     NaN           NaN
981        G     Pendiente
6730     NaN           NaN
2278     NaN   No indicado
1443     NaN   No indicado
2396     NaN   No indicado

In [54]:
#  Muestreo de valores unicos 
df_131["tipologia"].unique()

array(['Piso', 'Apartamento', 'Loft', 'Estudio', 'Duplex', 'Chalet',
       'Atico', 'Adosado', 'Negocio', 'Villa', 'Bungalow', 'Pareado',
       'Casa', 'Ático', 'Dúplex', 'Casa Individual', 'Casa Adosado',
       'Venta', 'Casa Pareado', 'Chalet Adosado', 'Chalet Pareado'],
      dtype=object)

In [55]:
# Muestreo de valores unicos
df_131["propiedad_completa"].unique()

array([nan,  1.])

In [56]:
# Muestreo de valores unicos 
df_131["estado"].unique()

array([nan, 'Para reformar', 'Bueno / Habitable',
       'Nuevo / En construcción', 'Óptimo / Reformado', 'En buen estado',
       'A estrenar', 'Reformado', 'A reformar', 'Estado Perfecto Estado',
       'Estado A reformar', 'Estado Depende del precio',
       'Estado Obra nueva'], dtype=object)

In [57]:
# Muestreo de valores unicos 
df_131["amueblado"].unique()

array(['0.0', '1.0', nan, 'Sólo cocina amueblada', 'Sí',
       'Parcialmente amueblado', 'No', 0.0, 1.0], dtype=object)

In [58]:
#  Seleccion de columnas y filtro por indice
filtro_planta= df_131[["planta","url"]]
filtro_planta["url"][7762]

'https://www.pisos.com/comprar/piso-rios_rosas28003-47513256186_108900/'

In [59]:
# Muestreo de valores aleatorios
df_131["planta"].sample(10)

1995    Planta baja
3229              2
6357            4.0
6607            3.0
2220            NaN
427             NaN
66              NaN
1729    Planta baja
6205            1.0
754             NaN
Name: planta, dtype: object

In [60]:
#  Muestreo de valores unicos 
df_131["habitaciones"].unique()

array([nan, '1', '2', '5', '4', '3', '5+'], dtype=object)

In [61]:
#  Muestreo de valores unicos
df_131["dormitorios"].unique()

array([ 3.,  4.,  5.,  1.,  2.,  0.,  6.,  7.,  9.,  8., 10., nan, 11.,
       12., 13., 24., 15.])

In [62]:
#  Filtro de valores mayor o igual a 20  
dormitorios_filtro=df_131[df_131["dormitorios"]>=20]
dormitorios_filtro["url"]

3045    https://www.indomio.es/anuncios/96701987/
3940    https://www.indomio.es/anuncios/96901535/
Name: url, dtype: object

In [63]:
#  Muestreo de valores unicos 
df_131["banos"].unique()

array(['1', '2', '3', '0', '4', '11', '5', '6', '7', '10', '31', nan,
       '3+', '1.0', '2.0', '59.0', '4.0', '5.0', '3.0', '32.0', '6.0',
       '270.0', '109.0', '146.0', '150.0', '110.0', '142.0', '445.0',
       '8.0', '5520.0', '333.0', '4661.0', '90.0', '7.0', '454.0', '9.0',
       '12.0', '10.0', '13.0', '11.0', '1330.0', '1390.0', '40.0', '35.0',
       '43.0', '38.0', '31.0', '47.0', '37.0', '17.0', '51.0', '46.0',
       '49.0', '23.0', '28.0', '48.0', '29.0', '96.0', '45.0', '95.0',
       '74.0', '66.0', '68.0', '30.0', '34.0', '3562.0', '278.0', '70.0',
       '130.0', '201.0', '85.0', '6642.0', '72.0', '220.0', '385.0',
       '50.0', '67.0', '82.0', '265.0', '397.0', '153.0', '76.0', '182.0',
       '119.0', '6413.0', '141.0', '57.0', '258.0', '75.0', '80.0',
       '3823.0', '285.0', '363.0', '217.0', '4444.0', '319.0', '214.0',
       '64.0', '69.0', '145.0', '161.0', '87.0', '118.0', '65.0', '97.0',
       '231.0', '170.0', '228.0', '175.0', '190.0', '440.0', '25.0

In [ ]:
#  Filtro de valores igual a 397.0 ....Aqui se puede ver claramente que en la descripcion menciona 3 baños y en la columna de banos aparecen 397.0.
banos_filtro= df_131[df_131["banos"]=="397.0"] 
banos_filtro

title  \
6320  Piso en venta en Castellana   

                                                                   url  \
6320  https://www.pisos.com/comprar/piso-goya28006-41702397465_102100/   

         precio  precio_anterior descuento  EUR/m2  m2_constr   distrito  \
6320  2500000.0              NaN       NaN  6297.0     6297.0  Salamanca   

      cod_distrito barrio  cod_barrio zona calle  \
6320           4.0   Goya        42.0  NaN   NaN   

                                                                                                                                                                                                                detalle  \
6320  Piso en castellana de 397 m², 3 baños, terraza de 65,00 metros, exterior, luminoso  características:  calefacción: central aire acondicionado ascensor portero automático/fisico estado de conservación: reformar   

     consumoce_ano letrace emisiones_co2 tipologia  propiedad_completa estado  \
6320           NaN     NaN           NaN      Piso                 NaN    NaN   

     amueblado planta habitaciones  dormitorios  banos  balcon  terraza  \
6320       NaN    NaN          NaN          3.0  397.0     0.0      1.0   

     garaje  trastero  ascensor porteria  
6320    NaN       0.0       1.0      NaN

In [65]:
# filtro por url
banos_filtro["url"]

6320    https://www.pisos.com/comprar/piso-goya28006-41702397465_102100/
Name: url, dtype: object

In [66]:
#  Filtro directo al valor "278.0" . Entrando al link se puede apreciar que el valor en si mismo no hace referencia a la cantidad de banos sino al espacio
# total del inmueble o llamado el pagina como superficie construida.
banos_filtro= df_131[df_131["banos"]=="278.0"] 
banos_filtro

title  \
5551  Piso en venta en Calle de San Bernardo, cerca de Calle Gran Vía   

                                                                             url  \
5551  https://www.pisos.com/comprar/piso-centro_palacio28013-48400895232_100200/   

         precio  precio_anterior descuento  EUR/m2  m2_constr distrito  \
5551  1823000.0              NaN       NaN  6557.0        2.0   Centro   

      cod_distrito   barrio  cod_barrio zona calle  \
5551           1.0  Palacio        11.0  NaN   NaN   

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              detalle  \
5551  Agencia inmobiliaria de madrid zona san bernardo - conde duque - malasaña oficina tecnocasa vende:\n-edificio construido en el año 1890\n-vivienda exterior con 1 mirador y 10 balcones todos ellos con vistas a la gran vía y calle san bernardo.\n-inmueble muy luminoso situado en la segunda planta del edificio compuesto actualmente por amplio hall de entrada con dos ventanales a patio de manzana y cocina, 9 dormitorios, 8 de ellos exteriores, chimenea, pasillo, distribuidor y 6 baños varios de ellos con ventana a patio de manzana.\n-la finca dispone de ascensor, portero físico y gas natural.\n-el inmueble necesita reforma completa con amplias posibilidades de distribución.\n-ubicación en pleno centro de madrid a escasos metros de la gran vía remodelada recientemente con su amplitud de aceras, rodeado del mejor ocio de madrid, bares, teatros, cines, tiendas de moda, terrazas y supermercados.\n-puntos de interés cercanos al inmueble:\ncalle gran vía\ncalle preciados\nplaza de callao\nplaza de españa\nplaza mayor\nplaza de santo domingo\nplaza de cibeles\npalacio real\nmercado de los mostenses con parking publico\ncalle luna con parking publico tudescos\n-lineas de metro cercanas, santo domingo l 2, noviciado l 2, plaza de españa\nl 10, gran via l 1 y l 5, tribunal l 1, con varias lineas disponibles de autobús, con\nestaciones de tren próximas, recoletos, sol y príncipe pio.   

     consumoce_ano letrace emisiones_co2 tipologia  propiedad_completa  \
5551           NaN     NaN           NaN      Piso                 NaN   

              estado amueblado  planta habitaciones  dormitorios  banos  \
5551  En buen estado       NaN  6557.0          NaN          9.0  278.0   

      balcon  terraza garaje  trastero  ascensor porteria  
5551     0.0      0.0    NaN       0.0       0.0      NaN

In [67]:
# Muestreo de valores unicos 
df_131["balcon"].unique()

array([nan,  1.,  0.])

In [68]:
# Muestreo de valores unicos
df_131["terraza"].unique()

array([nan,  1.,  0.])

In [69]:
# Muestreo de valores unicos
df_131["garaje"].unique()

array(['0.0', '1.0', nan, '2 en garaje privado',
       '1 en aparcamiento/garaje público', '1 en garaje privado',
       '1 en garaje privado, 3 en aparcamiento/garaje público',
       '5 en aparcamiento/garaje público',
       '2 en aparcamiento/garaje público',
       '1 en garaje privado, 1 en aparcamiento/garaje público',
       '4 en aparcamiento/garaje público', '8 en garaje privado',
       '1 en garaje privado, 5 en aparcamiento/garaje público',
       '6 en aparcamiento/garaje público',
       '3 en aparcamiento/garaje público',
       '9 en aparcamiento/garaje público',
       '10 en aparcamiento/garaje público',
       '2 en garaje privado, 4 en aparcamiento/garaje público',
       '1 en garaje privado, 2 en aparcamiento/garaje público',
       '2 en garaje privado, 2 en aparcamiento/garaje público', '2.0',
       '3.0', 'Garajes 1', 'Garajes 3', 'Garajes 2'], dtype=object)

In [70]:
# Muestreo de valores aleatorios en dos columnas
df_131[["garaje","url"]].sample(10)

garaje  \
5354         NaN   
6879         NaN   
2104         NaN   
6977         NaN   
10057  Garajes 1   
3205         NaN   
9735         NaN   
5525         NaN   
5409         NaN   
5302         NaN   

                                                                                                                                  url  
5354                                          https://www.pisos.com/comprar/piso-centro_embajadores_lavapies28012-42593492030_100200/  
6879                                                        https://www.pisos.com/comprar/piso-latina_aluche28047-48386743105_108900/  
2104                                                                                        https://www.indomio.es/anuncios/96366037/  
6977                                                         https://www.pisos.com/comprar/piso-lista_barrio28006-45928774561_100500/  
10057          https://www.redpiso.es/inmueble/chalet-en-venta-en-calle-gualda-rejas-san-blas-canillejas-madrid-madrid-RP322023116157  
3205                                                                                        https://www.indomio.es/anuncios/96750125/  
9735   https://www.redpiso.es/inmueble/piso-en-venta-en-calle-ronda-del-sur-entrevias-puente-de-vallecas-madrid-madrid-RP172023120847  
5525                                                       https://www.pisos.com/comprar/piso-centro_palacio28013-44217243661_996542/  
5409                                          https://www.pisos.com/comprar/piso-centro_embajadores_lavapies28012-48428351733_100200/  
5302                                                          https://www.pisos.com/comprar/piso-castillejos28046-37526521237_101800/

In [71]:
# Muestreo de valores unicos
df_131["trastero"].unique()

array([ 1.,  0., nan])

In [72]:
# Muestreo de valores unicos
df_131["ascensor"].unique()

array([nan,  1.,  0.])

In [73]:
# Muestreo de valores unicos 
df_131["porteria"].unique()

array([nan, 'Portero media jornada', 'Portero todo el día'], dtype=object)

### **_Propuestas_**

In [74]:
# nulos totales por columna...Aqui puse esto para pasarlo mas facil a la tabla
df_131.isna().sum()

title                  1470
url                       0
precio                    0
precio_anterior       10186
descuento             10186
EUR/m2                 2381
m2_constr                28
distrito                177
cod_distrito              9
barrio                  419
cod_barrio               77
zona                   4812
calle                  7800
detalle                4227
consumoce_ano          7091
letrace                8930
emisiones_co2          7091
tipologia                 0
propiedad_completa    10128
estado                 4666
amueblado              7371
planta                 4564
habitaciones           7175
dormitorios             187
banos                   296
balcon                 4383
terraza                4240
garaje                 8201
trastero               1950
ascensor               2222
porteria              10092
dtype: int64

In [ ]:
# nulos totalesen porcentaje por columna ... Aqui puse esto para pasarlo mas facil a la tabla
df_131.isna().mean()*100

title                 14.354067
url                    0.000000
precio                 0.000000
precio_anterior       99.462943
descuento             99.462943
EUR/m2                23.249683
m2_constr              0.273411
distrito               1.728347
cod_distrito           0.087882
barrio                 4.091397
cod_barrio             0.751880
zona                  46.987599
calle                 76.164437
detalle               41.275266
consumoce_ano         69.241285
letrace               87.198516
emisiones_co2         69.241285
tipologia              0.000000
propiedad_completa    98.896592
estado                45.561957
amueblado             71.975393
planta                44.565960
habitaciones          70.061517
dormitorios            1.825994
banos                  2.890343
balcon                42.798555
terraza               41.402207
garaje                80.080070
trastero              19.041109
ascensor              21.697100
porteria              98.545064
dtype: f

|Columnas|Descripcion contenido|Propuesta de cambios|Cant.Nulos|Propuesta de manejo de nulos|
|--------|---------------------|--------------------|----------|----------------------------|
|`title`   |Titular del lugar en venta. se puede encontrar en algunos casos la ciudad,tipo de inmueble, distrito, espacio m2,calle, entre otros valores.|1_era.Dejarla como esta  y hacer un ultimo filtro de valores donde no hayan valores en otras columnas y se puedan tomar de esta columna....2da_Hacer un ultimo filtro de los valores y eliminar la columna. Nota:habian algunos valores que se podian filtrar|1470 |Reeemplazar nulos con la palabra. sin_descripcion.|
|`url`     |link directo a la web perteneciente del anuncio. Incluye nombre de empresa, en algunas ocasiones el distrito, barrio, ciudad y numero identificador en el link| Sin cambios. Mantenerla ahi guardada| 0 |No aplica|
|`precio`  | Valor del inmueble| Eliminar las filas con valores igual a 1. Filas 774 y 1667.|0 |No aplica|
|`precio_anterior`|Valor anterior del inmueble sin descuento|Eliminarla |10186|ELiminar la columna, el 99.46% son nulos o dejarlo y rellenarlo con 0/-1 y /i usamos -1 recordar que algunos modelos no aceptan valores negativo.|
|`descuento`| cantidad percentil que se disminuye al precio antiguo|Eliminarla/Si se deja: Eliminar el signo porcentaje,agregarselo a la columna.|10186|ELiminar la columna, el 99.46% son nulos o dejarlo y rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo 
|`EUR/m2`  | valor en euro por metro cuadrado| Se detectaron valores anormales en relacion al precio por metro cuadrado, Hay algunos valores que se arreglan recorriendo una posicion el punto, otros agregandole 3 ceros y otros simplemente no dan con nada y tampoco hacen referencia a algo en especial de la propiedad despues de checarlo directamente en el link.Ejemplo: En la fila 8749 despues de checar la info del link se puede notar que el valor del euro por metro cuadrado no concuerda ni con ayuda.En la fila 893 se puede notar que al agregarle 3 ceros al valor por metro cuadrado al multiplicarlo por los metros cuadrados da resultado el precio.  Se noto que esos valores corresponden a pisos.com e indomio. 1er_propuesta: Eliminar la columna, arreglar la columna de m2_constr y con el precio y esa columna haciendo una division salen los valores del costo en euros por metro cuadrado. 2_propuesta: Darse el tiempo para checar esos valores atacandolos directamente. Viendo algun patron de los que les pueden agregar los 3 ceros, a los que les sirve recorrer un punto y eliminar los que no se ajusten para despues sacar el valor con las columnas precio y m2_constr.|2381/23.24%|rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo. Nota: se encontro 1 valor igual a 0|
|`m2_constr`|Espacio del inmueble en metros cuadrados y espacio ya construido de algunos inmuebles en metros cuadrados. |Parece que se mezclaron dos columnas con valores referentes a distintas columnas.Analizar valores anteriores antes de la union de la informacion de las diferentes paginas. |28|rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo. Nota: se encontro 4 valores igual a 0|
|`distrito`| Distritos| Quitar espacios, eliminar acentos.Normalizar el texto para que se unan distritos que deberian ser uno.|177/1.7%|Rellenarno con la palabra: sin_distrito.|
|`cod_distrito`| numero identificador por distrito| Asignar codigo de distrito a aquellos que tienen 0 y tienen distrito|9| Rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo|
|`barrio`  | barrio| normalizar valores para que se unan los que deberian ser uno.Nota:Se encontro cuatro caminos sin espacio entre ambas palabras y con espacio. Tambien una palabra llamada: lista|419|Rellenarlo con: sin_barrio.
|`cod_barrio`| numero identificador por barrio| Asignar codigo de barrio a quellos que tienen 0 y tienen barrio. Si no tiene codigo el barrio asginarle un nuevo codigo. Los otros 0 que no tengan barrio trabajarlo de la misma forma que los nulos|77|Rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo| 
|`zona`   |Contiene barrios,distritos, zonas entre dos barrios, zonas normales y residenciales, parques.| Usarla para hacer el filtro de los barrios y distritos faltantes y despues reemplazar esos valores con nulos para que solo queden las zonas y no barrio ni distrito.|4812/46.98%|Rellenarlo con: sin_zona.|
|`calle`  |Contiene el nombre de calles .|1_er propuesta. Dejar solo el nombre de la calle y eliminar lo demas para despues usar la calle y rellenar los barrios y distritos que se pueda con la calle(Se investigo y posiblemente el ayuntamiento de madrid pueda tener algunos datos abiertos con los barrio correspondientes a las calles , es solo una posibilidad.).2_da propuesta. Eliminar la columna si solo usaremos barrio y distrito.|7800/76.16%|Rellenarla con: sin_calle.| 
|`detalle`|Descripcion amplia sobre el inmueble, contiene informacion muy variada: espacio del lugar, barrio, distrito, numero de contacto, amueblado,puerta blindada, portero entre otros.| Hacer una ultima filtracion de informacion para rellenar donde no tengamos informacion pero podamos usar la que hay ahi y despues eliminarla.|4227/41.27%| Rellenarla con: sin_descripcion.|
|`consumoce_ano`|parecen representar datos relacionados con algún tipo de consumo, tal vez energético.|Tratar unidades inconsistentes como desconocido, ≥ 3,51.Identificar a que se refiere correctamente, su al consumo de energia o algun otro consumo. Nota: hay 2922 valores desconocidos, basicamente nulos. |7091 69.24% + 2922 desconocidos.|Rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo.
|`letrace`| Letra que hace referencia a la calificacion energetica. En el orden del abecederario empezamos con la "A" como la calificacion mas eficiente y se termina en la "G" como la mas ineficiente.|Sin modificacion|8930/87.19%| Rellenar los nulos con: sin_calificacion|
|`emisiones_co2`|Hace referencia a cantidad de kilogramos en emision de dioxido de carbono por metro cuadrado cada año. Gran parte de los datos vienen acompañados con la letra que hace referencia a la calificacion energetica| Segun la Directiva 2010/31/UE del Parlamento Europeo y del Consejo sobre la eficiencia energética de los edificios. Algunos de las calificaciones energeticas no son correctas. Hay que ajustar las calificaciones segun conrresponda correctamente. Filtrar las letras que hayan ahi a las filas de la columna letrace donde no hayan valores y despues solo dejar la cantidad de emision de co2 por m2 al año. Tambien cambiar el nombre de la columna a emisiones_kgCO₂/m²_año para solo dejar los numeros en las filas y finalmente trabajar los valores que no se refieren a las emisiones sino a palabras como exento, desconocido,entre otros.|7091/69.24%| Si se hace el cambio de nombre en la columna: Rellenarlo con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo. Si no se hace el cambio: rellenarno con : sin_registro.
|`tipologia`| Tipo de inmueble|Normalizar y reemplazar la palabra  "venta" por nulo que en si misma no hace referencia al tipo de inmueble y es vista como un tipo de |0|No aplica por ahora. Si se generan mas adelante relleñarlo con : sin_tipo
|`propiedad_completa`|  Hace referencia a si se  vende la propiedad completa o no| rellenar los nulos con 0 y dejar la columna en binario.Momentaneamente dejar la columa y analizar importancia con las demas columnas al igual que cuando se haga el feature importance.|10128/98.89%| Rellenar los nulos con 0|
|`estado`  | Hace referencia al estado del inmueble|Ajustar tipologia de estado a : reformar, buen estado, nuevo, reformado, estrenar, perfecto estado, depende precio|4666/45.56%|Rellear los nulos con: sin_registro.|
|`amueblado`| Hace referencia a si la propiedad se encuentra con muebles en alguna porcion, seccion o por completo.|Dejarlo en solo 4 categorias:1er_propuesta...solo cocina, No, si, parcialmente. 2da_propuesta... convertirla en dos columnas: una llamada amueblado y otra parcial_amueblado convertimos si donde sea parcialmente amueblado y donde la cocina este amueblada asi tenemos dos columnas binarias y no 1 categorical que luego necesitamos convertir en binario para algun proceso.|7371/71.97%| rellenarno con cero si hacemos las dos columnas y si solo queda una con la palabra : no o sin_muebles.
|`planta`  |En una parte hace referencia al numero de planta en el que se encuentra el inmueble. En otra parte se pueden encontrar encontrar numeros muy altos que hacen referencia a el costo en euros por metros metros cuadrados del inmueble. Se puede confirmar visualizando la fila 7762 con un valor 5759 donde se podra apreciar en el anuncio que se muestra 5.759 euros/m2|1er_propuesta...Arreglar los valores, analizando los pasos anteriores a la union de datos. Y asignarle una planta a los valores plante baja, intermedio y aquellos que se describen en texto sin planta precisa.2da_propuesta...Eliminar la columna|4564/44.56%|Si nos quedamos con la columna rellenar los nulos con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo|
|`habitaciones`| Hace referencia a la cantidad de espacios en el inmueble...salon,oficina, cocina en algunos casos, almacenamiento, se desconoce si tambien contiene los dormitorios principales o de huespedes| No aplica|7175/70.06%| Rellenarlo con : sin_registro.|
|`dormitorios`|Hace referencia a la cantidad de dormitorios(espacio destinado a descansar y dormir)| No aplica|187/1.82%| Rellenar los nulos con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo|
|`banos`   | En una parte hace referencia a los espacios de aseo personal. En otra parte hace referncia al espacio total del inmueble o tambien llamado en la pagina donde se encontro el dato como espacio construido. Se puede checar en la fila 5551 o filtrar por el valor "278.0". Pisos.com| Ajustar los valores adecuados, regresar a los pasos anteriores de la union o buscar una solucion que nos ayude a filtrar y generar los cambios necesarios.| 296/2.89%|
|`balcon`  |Hace referencia a si el inmueble cuenta o no con balcon( espacio exterior pequeño que sobresale de la estructura de un edificio)| No aplica|4383/%42.79| Rellenar los nulos con 0|
|`terraza` |Hace referencia a si el inmueble tiene terraza o no. Indica 0 cuando no y 1 cuando si.| No aplica|4240/41.40%| Rellenar los nulos con 0|
|`garage`  |Hace referencia a la cantidad e garajes del inmueble y si es publico, privado o tiene ambos.|1_er propuesta... Ajustar las categorias a nombres mas cortos. 2da_propuesta... Dividirla en dos columnas: cant_garage_privado y cantidad_garage_publico.|8201/80.08%| Si no se hace la division rellenarlo con la palabra: sin_garage. Si se hace la division:Rellenar los nulos con 0/-1 y si usamos -1 recordar que algunos modelos no aceptan valores negativo|
|`trastero`|Hace referencia a si el inmueble cuenta con trastero o no| No aplica| 1950/19.04%| Rellenar los nulos con 0|
|`ascensor`|Hace referencia a si el inmueble cuenta con ascensor| No aplica|2222/21.69%| Rellenarlo con 0|
|`porteria`|Indica si el inmueble cuenta con portero y si es en media o jornada completa| Ajustar a que solo indique si cuenta o no con portero. Convirtiendo la columna en binario| 10092/98.54%| Rellenar los nulos con 0 si la columna se convierte en binario. De lo contrario rellenarlos con la palabra: sin_portero 